packages/tools/src/utilities/contours/PointsArray.ts

```diff
@@ -0,0 +1,151 @@
+import type { Types } from '@cornerstonejs/core';
+import { vec2, vec3 } from 'gl-matrix';
+
+export type PolyDataPointConfiguration = {
+  dimensions?: 2 | 3;
+  initialSize?: number;
+};
+
+/**
+ * PointsArray is a TypedArray based representation of 2 or 3d points with
+ * custom sub-classes to represent the specific 2 or 3d values and to extract
+ * values as Types.Point2/Point3 values.
+ * This representation is efficient for storing large numbers of points and for
+ * transferring them amongst systems.
+ */
+export abstract class PointsArray<T> {
+  data: Float32Array;
+  dimensions = 3;
+  public length = 0;
+
+  constructor(configuration: PolyDataPointConfiguration = {}) {
+    const { initialSize = 1024, dimensions = 3 } = configuration;
+    this.data = new Float32Array(initialSize * dimensions);
+    this.dimensions = dimensions;
```

Comment:
shouldn't you update the legnth here as well?

Reply:
No, there isn't any data in it yet.

Reply:
You have data which is all zero